cloud/azure/locals.tf

```
@@ -0,0 +1,4 @@
+locals {
+  sfx_suffix               = var.sfx_integration_name_suffix == "" ? "" : format("-%s", var.sfx_integration_name_suffix)
+  default_integration_name = format("AzureSignalFxIntegration%s", local.sfx_suffix)
```

I'd leave the choice of the suffix separator `-` to the user of the module:

```suggestion
  default_integration_name = format("AzureSignalFxIntegration%s", var.sfx_integration_name_suffix)
```